package/MDAnalysis/auxiliary/XVG.py

```diff
@@ -0,0 +1,245 @@
+        int
+            Total number of steps
+        """
+        return len(self._auxdata)
+
+    def read_all_times(self):
+        """ Get list of time of each step.
+
+        Returns
+        -------
+        list of float
+            Time of each step
+        """
+        return self._auxdata[:,self.time_col]
+
+
+class XVGFileReader(base.AuxFileReader):
+    """ Auxiliary reader to read (step at a time) from an .xvg file.
+
+    An alternative XVG reader which reads each step from the .xvg file as 
+    needed (rather than reading + storing all from the start), for a lower 
+    memory footprint.     
+    
+    Parameters
+    ----------
+    filename : str
+       Location of the file containing the auxiliary data.
+    **kwargs
+       Other AuxReader options.    
+
+    See Also
+    --------
+    :class:`XVGReader`
+
+    Note
+    ----
+    The default reader for .xvg files is :class:`XVGReader`.
+
+    The time of each step is assumed to stored in the first column (``time_col``
+    defaults to 0), in units of ps, and data is assumed to be time-ordered.
+    """
+
+    format = 'XVG-F'
+
+    def __init__(self, filename, **kwargs):
+        time_col = kwargs.pop('time_col', 0)
+        super(XVGFileReader, self).__init__(filename, time_col=time_col, 
+                                        **kwargs)
+
+        
+    def _read_next_step(self):
+        """ Read next recorded step in xvg file. """
+        line = self.auxfile.readline()
+        if line:
+            # xvg has both comments '#' and grace instructions '@'
+            while line.lstrip()[0] in ['#', '@']:
+                line = self.auxfile.readline()
+            # remove end of line comments
+            line_no_comment = line.split('#')[0]
+            self.step = self.step + 1
+            self._data = [float(i) for i in line_no_comment.split()]
+            if self.n_cols and len(self._data) != self.n_cols:
+                raise ValueError('Step {0} has {1} columns instead of '
+                                 '{2}'.format(self.step, len(self._data),
+                                             self.n_cols))
+        else:
+            self.go_to_first_step()
+            raise StopIteration
+
+
+    def count_n_steps(self):
+        """ Iterate through all steps to count total number.
+
+        Returns
+        -------
+        int
+            Total number of steps
+        """
+
+        if self.constant_dt:
+            # will have to iterate through all to built times list anyway
+            return len(self.read_all_times())
+        else:
+            # iterate here instead
+            self._restart()
+            count = 0
+            for step in self:
+                count = count + 1
+            return count
+
+    def read_all_times(self):
+        """ Iterate through all steps to build times list.
+
+        Returns
+        -------
+        list of float
+            Time of each step
+        """
+        self._restart()

```

Comment:
`_restart` is the equivalent `_reopen` in the trajectory readers (they both just position before the first step, whereas `_rewind` then reads the first step); I renamed it here since we're not always dealing with files (and per #871, even when we reopening the file handle is maybe not the best idea).